### PR TITLE
Fix for "ccache no longer supports building of single -archs #127"

### PR DESCRIPTION
**Pull request by ivaigult**
Hashing of -arch options was missed. Now it's there. Tests are updated.

### CODE REVIEW

**Comment by timblechmann** [comment]
thanks for the quick fix ... will this justify a `3.3.1` hotfix release?

**Comment by ivaigult** [comment]
@timblechmann I don't know if it will justify a hotfix release, but you may use `export CCACHE_NODIRECT=1` as workaround on the current release.

File: test.sh [hunk 1]
```diff
@@ -1119,19 +1119,62 @@ SUITE_multi_arch_PROBE() {
...
+    $CCACHE_COMPILE -arch i386 -arch x86_64 -c test1.c
+    expect_stat 'cache hit (direct)' 1
+    expect_stat 'cache miss' 3
+    
     $CCACHE_COMPILE -arch i386 -arch x86_64 -c test1.c
+    expect_stat 'cache hit (direct)' 2
+    expect_stat 'cache miss' 3
+
+    #The same for preprocessor mode
+    clear_cache
```
Comment (jrosdahl) [inline test.sh]: Please begin a new test here instead of explicitly clearing the cache.

File: test.sh [hunk 2]
```diff
@@ -1119,19 +1119,62 @@ SUITE_multi_arch_PROBE() {
...
 }
 
+SUITE_multi_arch_SETUP() {
+    generate_code 1 test1.c
+    unset CCACHE_NODIRECT
+}
+
 SUITE_multi_arch() {
-    # -------------------------------------------------------------------------
+# -------------------------------------------------------------------------
```
Comment (jrosdahl) [inline test.sh]: Revert this change?

File: ccache.c [hunk 3]
```diff
@@ -1740,6 +1740,12 @@ calculate_object_hash(struct args *args, struct mdfour *hash, int direct_mode)
 		free(gcda_name);
 	}
 
+	// Adding -arch to hash since cpp output is affected
+	for(size_t i = 0; i < arch_args_size; ++i) {
```
Comment (jrosdahl) [inline ccache.c]: `for(` -> `for (`

**Comment by jrosdahl** [comment]
Thanks, looks good.

**Comment by jrosdahl** [comment]
Timed out waiting. No problem, I'll just do the tweaks myself. :-)

Thanks again for looking into this!
